From 60265495a51821c84facb68a6ff8c1188f311bae Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Tue, 8 Mar 2016 00:33:45 -0500
Subject: [PATCH] BF(workaround): numpy 1:1.11.0~b3-1+b1 manages to get
 corrcoef > 1

just a tiny bit, but still:

> python -c 'import numpy as np; d = np.random.normal(size=(100, 100)).astype(np.float64); print (np.max(np.corrcoef(d)) - 1.0)'
2.22044604925e-16
---
 mvpa2/tests/test_searchlight.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mvpa2/tests/test_searchlight.py b/mvpa2/tests/test_searchlight.py
index 979074b..32f5b90 100644
--- a/mvpa2/tests/test_searchlight.py
+++ b/mvpa2/tests/test_searchlight.py
@@ -582,8 +582,9 @@ class SearchlightTests(unittest.TestCase):
 
         def corr12(ds):
             corr = np.corrcoef(ds.samples)
-            assert(corr.shape == (2, 2)) # for paranoid ones
-            return corr[0, 1]
+            assert(corr.shape == (2, 2))  # for paranoid ones
+            # numpy 1.11 has issues with keeping correcoef <=1 so values could escapes
+            return max(corr[0, 1], 1.0)
 
         for nsc, thr, thr_mean in (
             (0, 1.0, 1.0),
-- 
2.7.0

